Mostly house style changes to the dictionary style guide. (#46)
COMCIFS/comcifs.github.io: accepted/ddlm_dictionary_style_guide.md

# Style Guide for DDLm Dictionaries

> Version 1.3.2 2026-02-05

Reviewer: ```suggestion
Version 1.4.0 2026-02-05
```

Reviewer: One rule is being extended so the minor version number must be increased.

`loop indent`: 2
`loop align`: 10
`loop step`: 5

> /bin/bash: line 1: a: command not found

Reviewer: This seems unintentional.

> # Array item in loop starts at column 37 to maintain min whitespace

Reviewer: ```suggestion
# Array item in loop starts at column 35 to maintain min whitespace
```

    loop_
      _dictionary_valid.scope
      _dictionary_valid.option
      _dictionary_valid.attributes
         Dictionary  Mandatory      ['_dictionary.title'  '_dictionary.class'
                                     '_dictionary.version'  '_dictionary.date'
                                     '_dictionary.uri'
                                     '_dictionary.ddl_conformance'
                                     '_dictionary.namespace']
         Dictionary  Recommended    ['_description.text'
                                     '_dictionary_audit.version'
                                     '_dictionary_audit.date'

>                                      '_dictionary_audit.revision']

Reviewer: ```suggestion
         Dictionary  Mandatory    ['_dictionary.title'  '_dictionary.class'
                                  '_dictionary.version'  '_dictionary.date'
                                  '_dictionary.uri'
                                  '_dictionary.ddl_conformance'
                                  '_dictionary.namespace']
         Dictionary  Recommended  ['_description.text'
                                  '_dictionary_audit.version'
                                  '_dictionary_audit.date'
                                  '_dictionary_audit.revision']
```

Reviewer: The item replacement (3 items instead of 2) made the layout no longer valid. I think my suggestion should fix it (it seems to match the one in the DDLm reference dictionary).

### 3.1 Attribute-value pairs

Note the following rule assumes that no DDLm attributes are longer
than (`value col` - `text indent`) - `min whitespace`. The length of a
value includes the delimiters. The rules for attribute-value pairs
cover items from Set categories as well as items from single-packet
Loop categories.

1. DDLm attributes appear lower-cased at the beginning of a line after
   `text indent` spaces.
2. A value with character length that is less than or equal to
   (`line length` - `value col`) + 1 starts in column `value col`.
3. A value with character length that is greater than

>    (`line length` - `value col` + 1 and less than or equal to

Reviewer: ```suggestion
   (`line length` - `value col`) + 1 and less than or equal to
```

4. Each packet starts on a new line. The final packet is followed by a single
   blank line.
5. The first character of the first value of a packet is placed in column `loop align`.
6. Non-compound values that are longer than `line length - loop step`
   characters are presented as semicolon-delimited text strings.
7. Semicolon-delimited text strings in loops are formatted as for
   Section 2.1, except that they are indented so that the first
   non-blank,non-prefix character of each line aligns with the first alphabetic
   character of the data-name header, that is, the first non-blank
   character appears in column `text indent` + `loop indent` + 2.
8. If the number of looped attributes `n` > 1, values in packets are
   separated by `min whitespace` together with any white space
   remaining at the end of the line distributed evenly between the
   packet items.  The following algorithm achieves this:
    1. Find largest integer `p` such that no data values before packet item
      `p` on the current line contain a newline and the sum of the widths
      of the next `p` packet items, separated by `min whitespace`, is not greater
      than `line length`. Call this total width.

>     2. Calculate "remaining white space" as `floor[(line length - total width)/(p-1)]`.

Reviewer: I understand that square brackets are used here for readability, but they might be a bit confusing here given that `floor()` is a hypothetical function and that square brackets are more associated with arrays.

### 4.1  Front matter and definitions

> 1. The first line contains the `#\#CIF2.0` identifier with no trailing white space.

Reviewer: ```suggestion
1. The first line contains the `#\\#CIF_2.0` identifier with no trailing white space.
```

Reviewer: I think that `CIF2.0` here was originally referring to the format rather than the specific string, but both options seem ok. However, if we are using a specific string, a stricter formulation might be used, e.g. `The first line consists of/is equal to/etc.  the `#\\#CIF_2.0` identifier with no trailing white space.`

Also, keep in mind, that since this is a markdown file, the backslash character needs to be escaped to be properly rendered.

> 1. The first line consists of the `#\\#CIF2.0` identifier with no trailing white space.

Reviewer: ```suggestion
1. The first line consists of the `#\#CIF_2.0` identifier with no trailing white space.
```

Reviewer: It seems that in backticks, the backslash does not need to be escaped after all. Anyway, the underscore symbol after the '2' was missing.

|   1.2.2 | 2023-06-29 | Added rule 2.5 for enumeration formatting. |
|   1.3.0 | 2025-05-23 | Added the "Known implementations" section. |
|   1.3.1 | 2025-06-13 | Added rule 3.2.11 for the formatting of the `_method.expression` attribute in a loop context. |

> |   1.3.2 | 2026-02-05 | Added DICTIONARY_AUTHORS category, updated some examples; otherwise largely cosmetic to align with International Tables house style. |

Reviewer: ```suggestion
|   1.4.0 | 2026-02-05 | Specified the placement and attribute order of the DICTIONARY_AUTHORS category, updated some examples; otherwise largely cosmetic to align with International Tables house style. |
```
